### PR TITLE
Extract table printer from CLI

### PR DESCRIPTION
**Pull request by mislav**
This exposes a TablePrinter that can be used to print column-formatted tabular data to the terminal and the same data in TSV format to redirect to files or scripts.

TODO
- [x] more tests
- [x] API docs

### CODE REVIEW

File: example_gh_test.go [hunk 1]
```diff
@@ -162,3 +164,25 @@ func ExampleCurrentRepository() {
...
+
+	// add a field that will render with color and will not be auto-truncated
+	t.AddField("1", tableprinter.WithColor(red), tableprinter.WithTruncate(nil))
+	t.AddField("hello")
+	t.EndRow()
+	t.AddField("2")
+	t.AddField("world")
+	if err := t.Render(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
```
Comment (samcoe) [inline example_gh_test.go]: I think this is equivalent to `log.Fatal(err)`.

File: pkg/tableprinter/table.go [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+type fieldOption func(*tableField)
+
+type TablePrinter interface {
+	AddField(string, ...fieldOption)
+	EndRow()
+	Render() error
+}
+
+// WithTruncate sets the truncation function for the field. Pass nil to disable truncation for this value.
```
Comment (samcoe) [inline pkg/tableprinter/table.go]: If no `WithTruncate` function is specified for a field will the the field be truncated as necessary to fit the output width?

File: pkg/term/env.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+// IsTerminalOutput returns true if standard output is connected to a terminal.
+func (t Term) IsTerminalOutput() bool {
+	return t.isTTY
+}
+
+// ColorEnabled reports whether it's safe to output ANSI color sequences, depending on IsTerminalOutput and
+// environment variables.
+func (t Term) ColorEnabled() bool {
```
Comment (samcoe) [inline pkg/term/env.go]: Do we want to name this `IsColorEnabled` to match naming scheme of `IsTerminalOutput`?

**Review by samcoe — APPROVED** [review]
This looks great! Added just a couple non-blocking comments/questions.

File: pkg/tableprinter/table.go (hunk 2 above)
Comment (mislav) [reply pkg/tableprinter/table.go]: That's a good question! Every field is automatically truncated by default. I will try to point this out in the docs

File: pkg/term/env.go (hunk 3 above)
Comment (mislav) [reply pkg/term/env.go]: Good idea!

**Pull request merged**